config/stacks/shared.ts

```diff
@@ -44,6 +44,7 @@ const getEventBusConstructProps = (): EventBusProps => {
     vpcProps: vpcProps,
     lambdaSecurityGroupName: archiveSecurityGroupName,
     archiveBucketName: archiveBucketName,
+    enableBucketRetainPolicy: n === 'prod',
```

Comment:
Just a suggestion, perhaps if we could have it as switch-case pattern and pass the RemovalPolicy from the props would be better (e.g the `getEventSourceConstructProps`)? I think the one liner `if-else` is a bit tedious to read for a config file that splits across stages?

Reply:
fixed